### PR TITLE
Add first pass at straightforward data models

### PR DESCRIPTION
**Pull request by kylesykes**
First pass at adding the data models.  I don't currently recursively create objects for things deeper in each of the objects (the `DestinationMapping` under a `Connection` for example is just a `dict` and not parsed to a `Mapping` object.  This can be updated later as we figure out use cases.

### CODE REVIEW

**Comment by MattPitlyk** [comment]
#47

File: models/Connection.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            "CreatedDate": Timestamp,
+            "ModifiedDate": Timestamp,
+            "Type": string (Salesforce, CW..),
+            "ConnectionInfo": {"connectionstring": ..},
+            "Name": string,
+            "UID": UID,
+        }
+        
+        """
+        pass
```
Comment (kylesykes) [inline models/Connection.py]: Infer input type passed into class.

Create constructor statements which check if it's a string (then call from_json) or if it's a dict (then call from_dict) when creating.

Usage: `Connection(connection_json)` would call `Connection.from_json(connection_json)`
and `Connection(connection_dict)` would call `Connection.from_dict(connection_dict)`

File: models/Organization.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            Transfers	Array[Transfer]
+            Transfer History	Array[]
+            Download History	Array[]
+            Upload History	Array[]
+        }
+        """
+        pass
+    
+    def from_dict(self, org_dict: Dict):
+        self._org_dict = org_dict
```
Comment (MattPitlyk) [inline models/Organization.py]: Add `name` field. (Was missing in my mock up).

**Comment by MattPitlyk** [comment]
All object should have `CreatedDate` and `CreatedBy` fields.

File: models/Connection.py [hunk 3]
```diff
@@ -17,26 +18,51 @@ def __init__(self, *args, **kwargs):
...
+
+    def from_json(self, conn_json: str):
+        try:
+            self.from_dict(json.loads(conn_json))
+        except Exception as e:
+            print("Parameter 'conn_json' is not a valid JSON string: " + e)
+
+    def from_tuple(self, conn_tuple: Tuple):
+        (
+            self._conn_dict,
```
Comment (MattPitlyk) [inline models/Connection.py]: Where does `_conn_dict` come from?

File: models/DataMapping.py [hunk 4]
```diff
@@ -13,17 +13,39 @@ def __init__(self, *args, **kwargs):
...
+    def from_tuple(self, dm_tuple: Tuple):
+        (
+            self._uid,
+            self._organization,
+            self._name,
+            self._mapping,
+            self._created_date,
+            self._created_by,
+        ) = dm_tuple
+
```
Comment (MattPitlyk) [inline models/DataMapping.py]: Do you need to set `self._dm_dict` here so that it's available for `to_dict()`?

File: models/Organization.py [hunk 5]
```diff
@@ -15,21 +17,35 @@ def __init__(self, *args, **kwargs):
...
+        # self._transfer_history = org_dict.get("TransferHistory", None)
+        # self._download_history = org_dict.get("DownloadHistory", None)
+        # self._upload_history = org_dict.get("UploadHistory", None)
 
     def from_json(self, org_json: str):
         self.from_dict(json.loads(org_json))
 
+    def from_tuple(self, org_tuple: Tuple):
+        (self._uid, self._name, self._created_date) = org_tuple
+
```
Comment (MattPitlyk) [inline models/Organization.py]: Do you need to need set `self._org_dict` here so that it's available for `to_dict()`?

File: models/Transfer.py [hunk 6]
```diff
@@ -23,28 +24,53 @@ def __init__(self, *args, **kwargs):
...
+            self._source,
+            self._source_mapping,
+            self._destination,
+            self._destination_mapping,
+            self._start_date_time,
+            self._frequency,
+            self._record_filter,
+            self._active,
+        ) = transfer_tuple
+
```
Comment (MattPitlyk) [inline models/Transfer.py]: Do you need to set `self._transfer_dict` here so it's available for `to-dict()`?

File: models/User.py [hunk 7]
```diff
@@ -14,18 +14,30 @@ def __init__(self, *args, **kwargs):
...
+        self._name = user_dict.get("Name", [])
+        self._created_date = user_dict.get("CreatedDate", None)
+        self._organization = user_dict.get("Organization", None)
 
     def from_json(self, user_json: str):
         self.from_dict(json.loads(user_json))
 
+    def from_tuple(self, user_tuple: Tuple):
+        (self._uid, self._name, self._created_date, self._org) = user_tuple
+
```
Comment (MattPitlyk) [inline models/User.py]: Do you need to set `self._user_dict` here so it's available for `to_dict()`?

File: models/DataMapping.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
+import json
+from typing import Dict, Tuple
+
+
+class DataMapping:
+    def __init__(self, *args, **kwargs):
```
Comment (MattPitlyk) [inline models/DataMapping.py]: Missing `data` parameter.

File: models/Organization.py [hunk 9]
```diff
@@ -0,0 +1,53 @@
+import json
+from typing import Dict, Tuple
+
+
+class Organization:
+    def __init__(self, *args, **kwargs):
```
Comment (MattPitlyk) [inline models/Organization.py]: Missing `data` parameter.

File: models/Transfer.py [hunk 10]
```diff
@@ -0,0 +1,78 @@
+import json
+from typing import Dict, Tuple
+
+
+class Transfer:
+    def __init__(self, *args, **kwargs):
```
Comment (MattPitlyk) [inline models/Transfer.py]: Missing `data` parameter.

File: models/User.py [hunk 11]
```diff
@@ -0,0 +1,45 @@
+import json
+from typing import Dict, Tuple
+
+class User:
+    def __init__(self, *args, **kwargs):
```
Comment (MattPitlyk) [inline models/User.py]: Missing `data` parameter.

File: models/User.py (hunk 7 above)
Comment (kylesykes) [reply models/User.py]: Fixed

File: models/Connection.py [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+        except Exception as e:
+            print("Parameter 'conn_dict' is not a valid dict: " + e)
+
+    def from_json(self, conn_json: str):
+        try:
+            return self.from_dict(json.loads(conn_json))
+        except Exception as e:
+            print("Parameter 'conn_json' is not a valid JSON string: " + e)
+
+    def from_tuple(self, conn_tuple: Tuple):
```
Comment (MattPitlyk) [inline models/Connection.py]: Use `cursor.description` to get column names and create a dict from that.

**Comment by kylesykes** [comment]
Made a change to the expected structure of the tuple parameter in `from_tuple`.

It will now look like:
```(('UID', 1), ('Organization', 1), ('Name', 'SF to HUD'), ('MappingInfo', '{}'))```

which simplifies the creation of the dictionary in `from_tuple`.

**Comment by kylesykes** [comment]
Just went ahead and merged in the database stuff to this models PR because they are so tightly coupled.  After doing that I went ahead and made all the `get` functions return the appropriate model objects.

All create/delete functions also exist.

**Comment by kylesykes** [comment]
Ref #54

**Pull request merged**